extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java

```diff
@@ -19,6 +19,17 @@ public class QuartzRuntimeConfig {
     @ConfigItem(defaultValue = "QuarkusQuartzScheduler")
     public String instanceName;
 
+    /**
+     * The amount of time in milliseconds that a trigger is allowed to be acquired and fired ahead of its scheduled fire time.
+     */
+    @ConfigItem(defaultValue = "0")
+    public long batchTriggerAcquisitionFireAheadTimeWindow;
```

Comment:
Wouldnt't it be better for users and also more consistent with the rest of the properties if this where a `Duration`?

Reply:
It depends. I think that it's kind of reasonable to follow the way Quartz defines the property, i.e. "the amount of time in milliseconds" so that users can just copy and paste properties from the quartz config file. OTOH `Duration` is indeed an idiomatic way in Quarkus.

Reply:
`Duration` is also used [here](https://github.com/quarkusio/quarkus/blob/2.14.2.Final/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java#L59).

Reply:
> `Duration` is also used [here](https://github.com/quarkusio/quarkus/blob/2.14.2.Final/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java#L59).

Yes, but this config property is not passed directly to quartz. It's a `quarkus-quartz` -specific property.

Reply:
Sure, but it creates a suboptimal user experience to not be consistent

Reply:
> Sure, but it creates a suboptimal user experience to not be consistent

To be fair, I think someone wanting to configure this will be pretty much experienced with quartz so they may want to just use the config value they are already used to.

Reply:
Yeah, I understand. I'll leave this up to you guyes

Reply:
I'm not sure. This config property will be very likely set by advanced users who seek consistency with Quartz. Just my 2c. That said, I don't have a problem with `Duration`.